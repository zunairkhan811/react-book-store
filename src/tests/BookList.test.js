import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { render } from '@testing-library/react';
import MyBookList from '../components/BookList';

jest.mock('react-redux');
jest.mock('react-router-dom', () => ({
  useParams: jest.fn(),
}));
jest.mock('../redux/books/fetchApi', () => jest.fn());

describe('MyBookList Component', () => {
  beforeEach(() => {
    useDispatch.mockReturnValue(jest.fn());
    useSelector.mockReturnValue({
      data: [
        {
          item_id: 1,
          title: 'Book 1',
          author: 'Author 1',
          category: 'Category 1',
        },
        {
          item_id: 2,
          title: 'Book 2',
          author: 'Author 2',
          category: 'Category 2',
        },
      ],
    });
  });

  it('should render MyBookList component with type', () => {
    useParams.mockReturnValue({ type: 'Category 1' });
    const { container } = render(<MyBookList />);
    expect(container).toMatchSnapshot();
  });

  it('should render MyBookList component without type', () => {
    useParams.mockReturnValue({ type: null });
    const { container } = render(<MyBookList />);
    expect(container).toMatchSnapshot();
  });
});