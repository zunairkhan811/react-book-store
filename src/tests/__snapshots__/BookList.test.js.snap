// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBookList Component should render MyBookList component with type 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card-container"
    >
      <div
        class="info-div"
      >
        <div
          class="remove-div"
        >
          <h1>
            Category 1
          </h1>
          <h2>
            Book 1
          </h2>
          <p>
            Author 1
          </p>
          <button
            class="remove-btn"
            type="button"
          >
            Comments
          </button>
          <button
            class="remove-btn"
            type="button"
            value="Remove"
          >
            Remove
          </button>
          <button
            class="remove-btn"
            type="button"
          >
            Edit
          </button>
        </div>
        <div
          class="complete-div"
        >
          <div
            class="oval"
          />
          <span>
            <p>
              100%
            </p>
            <p>
              Completed
            </p>
          </span>
        </div>
      </div>
      <div
        class="update-div"
      >
        <h2>
          Current Chapter
        </h2>
        <h2>
          Chapter 17
        </h2>
        <button
          class="update-btn"
          type="button"
        >
          Update Progress
        </button>
      </div>
    </div>
    <div
      class="line"
    />
  </div>
</div>
`;

exports[`MyBookList Component should render MyBookList component without type 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card-container"
    >
      <div
        class="info-div"
      >
        <div
          class="remove-div"
        >
          <h1>
            Category 1
          </h1>
          <h2>
            Book 1
          </h2>
          <p>
            Author 1
          </p>
          <button
            class="remove-btn"
            type="button"
          >
            Comments
          </button>
          <button
            class="remove-btn"
            type="button"
            value="Remove"
          >
            Remove
          </button>
          <button
            class="remove-btn"
            type="button"
          >
            Edit
          </button>
        </div>
        <div
          class="complete-div"
        >
          <div
            class="oval"
          />
          <span>
            <p>
              100%
            </p>
            <p>
              Completed
            </p>
          </span>
        </div>
      </div>
      <div
        class="update-div"
      >
        <h2>
          Current Chapter
        </h2>
        <h2>
          Chapter 17
        </h2>
        <button
          class="update-btn"
          type="button"
        >
          Update Progress
        </button>
      </div>
    </div>
    <div
      class="card-container"
    >
      <div
        class="info-div"
      >
        <div
          class="remove-div"
        >
          <h1>
            Category 2
          </h1>
          <h2>
            Book 2
          </h2>
          <p>
            Author 2
          </p>
          <button
            class="remove-btn"
            type="button"
          >
            Comments
          </button>
          <button
            class="remove-btn"
            type="button"
            value="Remove"
          >
            Remove
          </button>
          <button
            class="remove-btn"
            type="button"
          >
            Edit
          </button>
        </div>
        <div
          class="complete-div"
        >
          <div
            class="oval"
          />
          <span>
            <p>
              100%
            </p>
            <p>
              Completed
            </p>
          </span>
        </div>
      </div>
      <div
        class="update-div"
      >
        <h2>
          Current Chapter
        </h2>
        <h2>
          Chapter 17
        </h2>
        <button
          class="update-btn"
          type="button"
        >
          Update Progress
        </button>
      </div>
    </div>
    <div
      class="line"
    />
    <form
      class="form-container"
    >
      <div
        class="form-heading"
      >
        <h1>
          ADD NEW BOOK
        </h1>
      </div>
      <div
        class="input-container"
      >
        <input
          id="title"
          placeholder="Book title"
          required=""
          type="text"
          value=""
        />
        <br />
        <input
          id="author"
          placeholder="Author"
          required=""
          type="text"
          value=""
        />
        <br />
        <select
          data-testid="category-select"
          id="categorybtn"
          name="categorybtn"
          required=""
        >
          <option
            value=""
          >
            Select Category
          </option>
          <option
            value="Science Fiction"
          >
            Science Fiction
          </option>
          <option
            value="Horror"
          >
            Horror
          </option>
          <option
            value="Action"
          >
            Action
          </option>
          <option
            value="Comedy"
          >
            Comedy
          </option>
          <option
            value="Documentary"
          >
            Documentary
          </option>
          <option
            value="Thriller"
          >
            Thriller
          </option>
          <option
            value="Adventure"
          >
            Adventure
          </option>
          <option
            value="Romantic"
          >
            Romantic
          </option>
        </select>
        <button
          class="add-book-btn"
          type="submit"
          value="Add Book"
        >
          Add Book
        </button>
      </div>
    </form>
  </div>
</div>
`;
